src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts

```diff
@@ -628,6 +698,48 @@ export class ListSettingWidget extends AbstractListSettingWidget<IListDataItem>
 			siblingInputPlaceholder: localize('listSiblingInputPlaceholder', "Sibling..."),
 		};
 	}
+
+	private renderInputBox(value: ObjectValue, rowElement: HTMLElement): InputBox {
+		const valueInput = new InputBox(rowElement, this.contextViewService, {
+			placeholder: this.getLocalizedStrings().inputPlaceholder
+		});
+
+		valueInput.element.classList.add('setting-list-valueInput');
+		this.listDisposables.add(attachInputBoxStyler(valueInput, this.themeService, {
+			inputBackground: settingsSelectBackground,
+			inputForeground: settingsTextInputForeground,
+			inputBorder: settingsTextInputBorder
+		}));
+		this.listDisposables.add(valueInput);
+		valueInput.value = value.data.toString();
+
+		return valueInput;
+	}
+
+	private renderDropdown(value: ObjectKey, rowElement: HTMLElement): SelectBox {
+		if (value.type !== 'enum') {
+			throw new Error('Valuetype must be enum.');
+		}
+		const selectBoxOptions = value.options.map(({ value, description }) => ({ text: value, description }));
+		const selected = value.options.findIndex(option => value.data === option.value);
+
+		const selectBox = new SelectBox(selectBoxOptions, selected, this.contextViewService, undefined, {
```

Comment:
Does it make sense to share any of this with the selectbox object value renderer?

Reply:
Yep. I created a helper method in `AbstractListSettingWidget`.